### ISSUE TITLE
Installation issues

### ISSUE THREAD

**Issue by sunabcun** [heathsc/gemBS-rs#3]
Hi, I have tried to install gemBS in multiple ways: gemBS, gemBS-rs, container images(like singularity), and Bioconda.

1. gemBS/gemBS-rs: failing compilation (I have attached parts of words from the errors when I tried gemBS-rs compiling.) , it seems to have a
[errors.txt](https://github.com/heathsc/gemBS-rs/files/6202972/errors.txt)
[errors.txt](https://github.com/heathsc/gemBS-rs/files/6202974/errors.txt)

 similar issue when it passed around bcf tools even in gemBS.

2. container image- it works but when I even tried to check the example run from the tutorials `gemBS prepare -c example.conf -t example.csv`. It says either example.conf does not exist or example.csv does not exist although I have run the code in the folder which has both files.

3. Bioconda download is consistently failed as well. I tried to install other programs like samtools etc., and all of them worked. I checked if it exists using conda search, and it showed that bioconda channel has gemBS but the connection was frozen and not moving forward. 

I'm very frustrated since I can't even compile gemBS or run the tutorial data set (the 1st step: "prepare" part).

**Comment by heathsc** [comment]
Sorry to hear about your problems with installation and testing.  From the error files you provide it looks as though you will need to install libssl and libcom_err to be able to finish compilation.

For the container issues - can you tell me which version of singularity you are using?  With earlier versions (2.x I think) it was often the case that only the home filesystem was mounted by default in the container, so if you ran it from a mounted filesystem (I see you were running on gpfs) it is possible that the files were not accessible.  In the instructions for the previous version of gemBS (v3) I included instructions for dealing with this issue (see [http://statgen.cnag.cat/gemBS/v3/UserGuide/_build/html/containers.html](http://statgen.cnag.cat/gemBS/v3/UserGuide/_build/html/containers.html))

**Comment by sunabcun** [comment]
Thank you for the reply.

I am using school’s HPC and they are not allowed to let me use sudo for installing it. So, I have tried now in my local setting but it still seems to have issues.
I have attached the error after installing all required things including libssl and libcom_err.
And, I will try it with singularity issue as well.

**Comment by heathsc** [comment]
It looks like you might not have rust installed in your local setting.  Have you tried installing with:

curl https://sh.rustup.rs -sSf > rust.sh && sh rust.sh -y

?

**Comment by sunabcun** [comment]
Thank you for suggesting. So, I previously installed the rust but I reinstalled again and it works until last moment and it made executable gemBS but it showed these errors:
Finished release [optimized] target(s) in 21m 29s
cp target/release/gem_bs bin/gemBS
cp target/release/read_filter bin/
cp target/release/bs_call bin/
cp target/release/snpxtr bin/
cp target/release/mextr bin/
cp target/release/dbsnp_index bin/
make[2]: Leaving directory '/home/yuna/gemBS-rs/rust'
**************************************************************************************************
copy shell script gemBS to a directory in your PATH (i.e., /usr/local/bin or /home/yuna/bin)
**************************************************************************************************
echo "#!/bin/sh" > gemBS
echo 'LD_LIBRARY_PATH=/usr/local/lib/gemBS/lib:$LD_LIBRARY_PATH /usr/local/lib/gemBS/bin/gemBS $@' >> gemBS
chmod 755 gemBS
mkdir -p /usr/local/lib/gemBS
mkdir: cannot create directory ‘/usr/local/lib/gemBS’: Permission denied
make[1]: *** [Makefile.gemBS:65: /usr/local/lib/gemBS] Error 1
make[1]: Leaving directory '/home/yuna/gemBS-rs'
make: *** [Makefile:7: install] Error 2

So, I tried to execute again with sudo make install. But, it was not working and I did delete whole gemBS_rs folder and re-install all the things again. And, I still get the similar empty container error and it was not resolved by installing rust again. Interestingly, if I execute “make install” not “ sudo make install”, I got the similar error again from the first attempt->

make[3]: Entering directory '/home/yuna/gemBS-rs/c_tools/htslib'
mkdir -p -m 755 /home/yuna/gemBS-rs/c_tools/tmp_root/bin /home/yuna/gemBS-rs/c_tools/tmp_root/include /home/yuna/gemBS-rs/c_tools/tmp_root/include/htslib /home/yuna/gemBS-rs/c_tools/tmp_root/lib /home/yuna/gemBS-rs/c_tools/tmp_root/share/man/man1 /home/yuna/gemBS-rs/c_tools/tmp_root/share/man/man5 /home/yuna/gemBS-rs/c_tools/tmp_root/share/man/man7 /home/yuna/gemBS-rs/c_tools/tmp_root/lib/pkgconfig
if test -n ""; then mkdir -p -m 755 ; fi
install -p -m 644 libhts.so /home/yuna/gemBS-rs/c_tools/tmp_root/lib/libhts.so.1.11
install: cannot remove '/home/yuna/gemBS-rs/c_tools/tmp_root/lib/libhts.so.1.11': Permission denied
make[3]: *** [Makefile:562: install-so] Error 1
make[3]: Leaving directory '/home/yuna/gemBS-rs/c_tools/htslib'
make[2]: *** [Makefile:43: /home/yuna/gemBS-rs/c_tools/tmp_root/include/htslib/hts.h] Error 2
make[2]: Leaving directory '/home/yuna/gemBS-rs/c_tools'
make[1]: *** [Makefile.gemBS:60: c_tools] Error 2
make[1]: Leaving directory '/home/yuna/gemBS-rs'
make: *** [Makefile:7: install] Error 2

I have attached the result(text files) from “sudo make install” after removing everything as well.

**Comment by heathsc** [comment]
By default the installation directory is /usr/local/lib/gemBS.  You can either run the install using sudo or change the installation directory by editing gemBS_config.mk.in  and re-running the install.

Simon

**Comment by sunabcun** [comment]
Hi Simon,
I have tried to install following your suggestion and it works. And, regarding the odd behavior in singularity run, it seems like my HPC-singularity has some issues with calling the program properly. It doesn’t have issues when I ran it in my local computer. I am asking my HPC people to install gemBS since I don’t have a permission but glad to know it doesn’t have an issue to install the program itself. Many thanks and I will close my ticket!
